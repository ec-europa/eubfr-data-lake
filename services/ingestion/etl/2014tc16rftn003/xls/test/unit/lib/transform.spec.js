/**
 * @jest-environment node
 */

import mapper from '../../../src/lib/transform';
import testRecord from '../../stubs/record.json';

describe('2014tc16rftn003 XLS transformer', () => {
  let result = {};

  beforeAll(() => {
    result = mapper(testRecord);
  });

  test('Returns null when record is not provided', () => {
    expect(mapper()).toBe(null);
  });

  test('Produces correct JSON output structure', () => {
    expect(result).toMatchSnapshot();
  });

  test('In field `third_parties`, respect countries of several words', () => {
    const copy = JSON.parse(JSON.stringify(testRecord));

    copy.Country =
      'FI Finland / SUOMI / FINLAND;UK United Kingdom / UNITED KINGDOM; NRC Not a real country / NONEXISTENT';
    const copyResult = mapper(copy);

    expect(copyResult.third_parties[0].country).toBe('Finland');
    expect(copyResult.third_parties[1].country).toBe('United Kingdom');
    expect(copyResult.third_parties[2].country).toBe('Not a real country');
  });
});
