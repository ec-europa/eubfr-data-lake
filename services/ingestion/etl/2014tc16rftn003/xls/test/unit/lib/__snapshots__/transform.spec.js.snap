// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2014tc16rftn003 XLS transformer Produces correct JSON output structure 1`] = `
Object {
  "action": "",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "EUR",
      "raw": 83,
      "value": 1564897.7617,
    },
    "funding_area": Array [],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "EUR",
      "raw": 1885418.99,
      "value": 1885418.99,
    },
  },
  "call_year": "",
  "complete": false,
  "description": "Priority axis: 1. Cooperating on innovation to make CENTRAL EUROPE more competitive
Operation summary: Healthy and active aging is one of the greatest social challenges in the EU in view of longer life expectancy. Early prevention of chronic diseases that have severe impact on general well-being of patients is extremely important if we want to reach this goal and ensure sustainable health care systems in central Europe. A lack of development and promotion of innovative quality health services, disease management models and education have a profound negative effect. 

To tackle this challenge, Focus in CD will demonstrate development and pilot testing of innovative health service model in management of celiac disease.
Programme & operation specific objective: To improve skills and entrepreneurial competences for advancing economic and social innovation in central European regions
Union co-financing rate in % (average as in CP): 83;83;83;83;83;83;83;83;83;83;83;83
",
  "ec_priorities": Array [],
  "media": Array [],
  "programme_name": "",
  "project_id": "cdb41bb3c99ff960bc0e16671d7b66ee",
  "project_locations": Array [
    Object {
      "address": "",
      "centroid": null,
      "country_code": "SI",
      "location": null,
      "nuts": Array [
        Object {
          "code": "SI0",
          "level": 1,
          "name": "",
          "year": null,
        },
        Object {
          "code": "SI03",
          "level": 2,
          "name": "",
          "year": null,
        },
        Object {
          "code": "SI032",
          "level": 3,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "2000",
      "region": "",
      "town": "Maribor",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "HU",
      "location": null,
      "nuts": Array [
        Object {
          "code": "HU1",
          "level": 1,
          "name": "",
          "year": null,
        },
        Object {
          "code": "HU10",
          "level": 2,
          "name": "",
          "year": null,
        },
        Object {
          "code": "HU102",
          "level": 3,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "1089",
      "region": "",
      "town": "Budapest",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "IT",
      "location": null,
      "nuts": Array [
        Object {
          "code": "ITH",
          "level": 1,
          "name": "",
          "year": null,
        },
        Object {
          "code": "ITH4",
          "level": 2,
          "name": "",
          "year": null,
        },
        Object {
          "code": "ITH44",
          "level": 3,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "34100",
      "region": "",
      "town": "TRIESTE",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "HU",
      "location": null,
      "nuts": Array [
        Object {
          "code": "HU1",
          "level": 1,
          "name": "",
          "year": null,
        },
        Object {
          "code": "HU10",
          "level": 2,
          "name": "",
          "year": null,
        },
        Object {
          "code": "HU102",
          "level": 3,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "1025",
      "region": "",
      "town": "Budapest",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "HR",
      "location": null,
      "nuts": Array [
        Object {
          "code": "HR0",
          "level": 1,
          "name": "",
          "year": null,
        },
        Object {
          "code": "HR03",
          "level": 2,
          "name": "",
          "year": null,
        },
        Object {
          "code": "HR031",
          "level": 3,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "51000",
      "region": "",
      "town": "Rijeka",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "HR",
      "location": null,
      "nuts": Array [
        Object {
          "code": "HR0",
          "level": 1,
          "name": "",
          "year": null,
        },
        Object {
          "code": "HR03",
          "level": 2,
          "name": "",
          "year": null,
        },
        Object {
          "code": "HR031",
          "level": 3,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "51000",
      "region": "",
      "town": "Rijeka",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "DE",
      "location": null,
      "nuts": Array [
        Object {
          "code": "DE2",
          "level": 1,
          "name": "",
          "year": null,
        },
        Object {
          "code": "DE21",
          "level": 2,
          "name": "",
          "year": null,
        },
        Object {
          "code": "DE212",
          "level": 3,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "80337",
      "region": "",
      "town": "München",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "SI",
      "location": null,
      "nuts": Array [
        Object {
          "code": "SI0",
          "level": 1,
          "name": "",
          "year": null,
        },
        Object {
          "code": "SI03",
          "level": 2,
          "name": "",
          "year": null,
        },
        Object {
          "code": "SI032",
          "level": 3,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "2000",
      "region": "",
      "town": "Maribor",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "DE",
      "location": null,
      "nuts": Array [
        Object {
          "code": "DE2",
          "level": 1,
          "name": "",
          "year": null,
        },
        Object {
          "code": "DE21",
          "level": 2,
          "name": "",
          "year": null,
        },
        Object {
          "code": "DE212",
          "level": 3,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "80337",
      "region": "",
      "town": "München",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "HR",
      "location": null,
      "nuts": Array [
        Object {
          "code": "HR0",
          "level": 1,
          "name": "",
          "year": null,
        },
        Object {
          "code": "HR03",
          "level": 2,
          "name": "",
          "year": null,
        },
        Object {
          "code": "HR031",
          "level": 3,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "51000",
      "region": "",
      "town": "Rijeka",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "SI",
      "location": null,
      "nuts": Array [
        Object {
          "code": "SI0",
          "level": 1,
          "name": "",
          "year": null,
        },
        Object {
          "code": "SI03",
          "level": 2,
          "name": "",
          "year": null,
        },
        Object {
          "code": "SI032",
          "level": 3,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "2250",
      "region": "",
      "town": "Ptuj",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "IT",
      "location": null,
      "nuts": Array [
        Object {
          "code": "ITH",
          "level": 1,
          "name": "",
          "year": null,
        },
        Object {
          "code": "ITH4",
          "level": 2,
          "name": "",
          "year": null,
        },
        Object {
          "code": "ITH44",
          "level": 3,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "34100",
      "region": "",
      "town": "Trieste",
    },
  ],
  "project_website": "",
  "related_links": Array [],
  "reporting_organisation": "Member states",
  "results": Object {
    "available": "",
    "result": "",
  },
  "status": "",
  "sub_programme_name": "",
  "success_story": "",
  "themes": Array [
    "(1b) promoting business investment in R&I",
    "developing links and synergies between enterprises",
    "research and development centres and the higher education sector",
    "in particular promoting investment in product and service development",
    "technology transfer",
    "social innovation",
    "eco-innovation",
    "public service applications",
    "demand stimulation",
    "networking",
    "clusters and open innovation through smart specialisation",
    "and supporting technological and applied research",
    "pilot lines",
    "early product validation actions",
    "advanced manufacturing capabilities and first production",
    "in particular in key enabling technologies and diffusion of general purpose technologies",
  ],
  "third_parties": Array [
    Object {
      "address": "",
      "country": "Slovenia",
      "email": "",
      "name": "Municipality of Maribor",
      "phone": "",
      "region": "",
      "role": "Leader",
      "type": "",
      "website": "",
    },
    Object {
      "address": "",
      "country": "Hungary",
      "email": "",
      "name": "Heim Pál Children’s Hospital",
      "phone": "",
      "region": "",
      "role": "Partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "",
      "country": "Italy",
      "email": "",
      "name": "University of Trieste",
      "phone": "",
      "region": "",
      "role": "Partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "",
      "country": "Hungary",
      "email": "",
      "name": "Hungarian Coeliac Society",
      "phone": "",
      "region": "",
      "role": "Partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "",
      "country": "Croatia",
      "email": "",
      "name": "Primorje - Gorski Kotar County",
      "phone": "",
      "region": "",
      "role": "Partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "",
      "country": "Croatia",
      "email": "",
      "name": "Association of celiac patients Primorsko - goranska County",
      "phone": "",
      "region": "",
      "role": "Partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "",
      "country": "Germany",
      "email": "",
      "name": "Child Health Foundation",
      "phone": "",
      "region": "",
      "role": "Partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "",
      "country": "Slovenia",
      "email": "",
      "name": "University Medical Center Maribor",
      "phone": "",
      "region": "",
      "role": "Partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "",
      "country": "Germany",
      "email": "",
      "name": "Ludwig-Maximilian’s University Medical Center",
      "phone": "",
      "region": "",
      "role": "Partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "",
      "country": "Croatia",
      "email": "",
      "name": "University hospital center Rijeka",
      "phone": "",
      "region": "",
      "role": "Partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "",
      "country": "Slovenia",
      "email": "",
      "name": "E-institute - Institute for Comprehensive Development Solutions",
      "phone": "",
      "region": "",
      "role": "Partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "",
      "country": "Italy",
      "email": "",
      "name": "Institute for Maternal and Child Health - IRCCS “Burlo Garofolo”",
      "phone": "",
      "region": "",
      "role": "Partner",
      "type": "",
      "website": "",
    },
  ],
  "timeframe": Object {
    "from": "2016-06-01T00:00:00.000Z",
    "from_precision": "day",
    "to": "2019-05-31T00:00:00.000Z",
    "to_precision": "day",
  },
  "title": "Focus IN CD",
  "type": Array [],
}
`;
